test/charms/test_secrets/src/charm.py

```diff
@@ -80,6 +81,54 @@ def add_with_meta(self, event: ops.ActionEvent):
         }
         event.set_results(cast('dict[str, Any]', result))
 
+    def set_secret_flow(self, event: ops.ActionEvent):
+        secretid = event.params.get('secretid')
+        secretlabel = event.params.get('secretlabel')
+        contentses = ({'val': f'{i}'} for i in itertools.count(1))
+        labels = (f'label{i}' for i in itertools.count(1))
+        descriptions = (f'description{i}' for i in itertools.count(1))
+        expires = (datetime.datetime(2010 + i, 1, 1, 0, 0, 0) for i in itertools.count(1))
+        rotates = iter(ops.SecretRotate.__members__.values())
```

Comment:
Since all the others are now infinite, I'd make this one infinite as well, via `itertools.cycle`. The assert should just need a `% length`, I would think.